node/src/main.js

```diff
@@ -21,51 +21,57 @@ const assertCommandExists = command => {
   }
 };
 
-const isInternalCommand = async (command, args) => {
-  let isInternalCmd = false;
+const hasHelpArg = args => ['-h', '--help'].some(h => args.includes(h));
 
-  if (['-h', '--help'].some(h => args.includes(h)) || command === 'help') {
-    help();
-    isInternalCmd = true;
-  }
+const hasVersionArg = args => ['--version'].some(h => args.includes(h));
 
-  if (['--version'].some(h => args.includes(h)) || command === 'version') {
-    logger.info(pkg.version);
-    isInternalCmd = true;
-  }
+const showVersion = () => {
+  logger.info(pkg.version);
+};
 
-  if (command === 'configure') {
-    await configure();
-    isInternalCmd = true;
-  }
+const runInternalCommand = async (command, options) => {
+  const commands = {
+    help: () => help(),
+    version: () => showVersion(),
+    configure: async () => configure(options)
+  };
 
-  return isInternalCmd;
+  await commands[command]();
 };
 
 const run = async () => {
   const mainOptions = commandLineArgs(mainDefinitions, { stopAtFirstUnknown: true });
   const argv = mainOptions._unknown || [];
 
-  const { command } = mainOptions;
+  let { command } = mainOptions;
 
   updateNotifier({ pkg }).notify();
 
   try {
-    if (await isInternalCommand(command, argv)) return;
+    if (hasVersionArg(argv)) command = 'version';
+    if (hasHelpArg(argv)) command = 'help';
 
     assertCommandExists(command);
 
-    const commandDefinitions = commands[command].options;
-    const commandsOptions = commandLineArgs(commandDefinitions, { argv });
+    const { options: commandDefinitions, internal, ignoreFlags } = commands[command];
 
-    const currentCommandOptions = commandsOptions[command];
+    let currentCommandOptions = {};
+    if (!ignoreFlags) {
```

Comment:
As discussed, `ignoreFlags` is quite dependant on `internal` and it makes this flow kinda unstable and hard to understand. I'd suggest to run w/e necessary explicitly for the `configure` command instead of using such flag.

Reply:
eb04c4f53f664a9ee2d25eaa490bf8b6d2702942